enhancements/dns/forward-dns-over-tls.md

```diff
@@ -0,0 +1,266 @@
+OpenShift.
+
+### Goals
+
+1. A cluster administrator can specify a CA certificate or CA bundle, TLS configuration, and an upstream FQDN for the default zone in the DNS configuration.
+2. A cluster administrator can specify a CA certificate or CA bundle, TLS configuration, and an upstream FQDN for a non-default zone.
+3. A cluster administrator can opt-in to DNS-over-TLS (using the configured CA/bundle) for each listed upstream in the default zone.
+4. A cluster administrator can opt-in to DNS-over-TLS (using the configured CA/bundle) for a set of upstreams on a non-default zone.
+
+### Non-Goals
+
+* Allowing configuration of DNS-over-TLS for use cases other than OpenShift's default DNS provider to upstream resolvers is out of scope.
+
+## Proposal
+
+This proposal covers changes to two APIs. The first is a change to 
+`UpstreamResolvers`.  To enable a cluster administrator to configure 
+DNS-over-TLS, the `UpstreamResolvers` type will get three new optional members:
+
+`TLSSecurityProfile` encapsulates TLS specific configuration. Including this 
+allows customers to set their desired TLS version, ciphers, etc.
+
+`CABundle` references a ConfigMap that contains a single CA certificate 
+or a CA bundle.
+
+`ServerName` contains an FQDN to validate when establishing the connection. 
+This is relevant in particular with cases where a customer has multiple 
+upstreams behind one serving certificate (e.g. `1.1.1.1` and `2.2.2.2` 
+behind `dns.myorg.com`)
+
+Since `UpstreamResolvers` doesn't allow for configuration of a custom zone, 
+it's assumed that the same TLS configuration will apply to _n_ `Upstream` 
+objects. 
+
+```go
+type UpstreamResolvers struct {
+    // ...snip
+
+    // TLSSecurityProfile encapsulates TLS configuration to use when
+    //establishing a connection with the upstream resolvers
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    TLSSecurityProfile v1.TLSSecurityProfile `json:tls,omitempty`
+
+    // CABundle references a ConfigMap that must contain either a single
+    // CA Certificate or a CA Bundle in the case of multiple upstreams signed
+    // by different CAs.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    CABundle *corev1.ConfigMap `json:caBundle,omitempty`
+	
+    // Server name to validate in the case of one or more upstreams behind one
+    // FQDN.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    ServerName string
+}
+```
+
+`Upstream` will get one new optional member:
+
+`UseTLS` which allows an individual upstream to be opted into using
+DNS-over-TLS. This will be useful in cases where not all upstreams support
+DNS-over-TLS.
+
+```go
+type Upstream struct {
+    // ...snip
+	
+    // UseTLS allows cluster administrators to opt-in to using the TLS config in
+    // UpstreamResolvers for upstreams that support DNS-over-TLS.
+    //
+    // +optional
+    // +kubebuilder:validation:Enum=;true;false
+    // +kubebuilder:default=false
```

Comment:
I switched this to `EnableTLS` as that makes sense to me. Seems like this should be explicitly opt-in since regular DNS queries should work by default but TLS queries may not work by default unless that configuration is correct and the upstream supports it.